### PR TITLE
AV1 - VA-API structure for rocdecode

### PR DESCRIPTION
**Pull request by LakshmiKumar23**

### CODE REVIEW

File: api/rocdecode.h [hunk 1]
```diff
@@ -782,6 +782,599 @@ typedef struct _RocdecHevcIQMatrix {
...
+    union {
+        struct {
+            /** Indicates whether segmentation map related syntax elements
+            *  are present or not for current frame. If equal to 0,
+            *  the segmentation map related syntax elements are
+            *  not present for the current frame and the control flags of
+            *  segmentation map related tables feature_data[][], and
+            *  feature_mask[] are not valid and shall be ignored by accelerator.
+            */
+            uint32_t         enabled                                     : 1;
```
Comment (AryanSalmanpour) [inline api/rocdecode.h]: Please remove extra spaces for this file's struct members, including the spaces between datatypes and variable names, spaces between variable names and equal signs, etc.
For example change `uint32_t               enabled                        : 1;` to 
`uint32_t enabled : 1;`

**Review by AryanSalmanpour — CHANGES_REQUESTED** [review]

File: api/rocdecode.h [hunk 2]
```diff
@@ -782,6 +782,599 @@ typedef struct _RocdecHevcIQMatrix {
     uint32_t                reserved[4];
 } RocdecHevcIQMatrix;
 
+/** \brief Segmentation Information for AV1
+  */
+typedef struct _RocdecSegmentationStructAV1 {
```
Comment (jeffqjiangNew) [inline api/rocdecode.h]: Maybe change to _RocdecAv1SegmentationStruct to be consistent with other namings.

File: api/rocdecode.h [hunk 3]
```diff
@@ -782,6 +782,599 @@ typedef struct _RocdecHevcIQMatrix {
...
+     *  Each bit field itself is the feature_id. Index is segment_id.
+     *  feature_mask[segment_id] & (1 << feature_id) equal to 1 specify that the feature of
+     *  feature_id for segment of segment_id is enabled, otherwise disabled.
+     */
+    uint8_t feature_mask[8];
+
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t reserved[4];
+
+} RocdecSegmentationStructAV1;
```
Comment (jeffqjiangNew) [inline api/rocdecode.h]: Maybe change to RocdecAv1SegmentationStruct to be consistent with other namings.

File: api/rocdecode.h [hunk 4]
```diff
@@ -782,6 +782,599 @@ typedef struct _RocdecHevcIQMatrix {
...
+    uint8_t cb_luma_mult;
+    uint16_t cb_offset;
+    uint8_t cr_mult;
+    uint8_t cr_luma_mult;
+    uint16_t cr_offset;
+
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t reserved[4];
+
+} RocdecFilmGrainStructAV1;
```
Comment (jeffqjiangNew) [inline api/rocdecode.h]: Similar suggestion as above.

File: api/rocdecode.h [hunk 5]
```diff
@@ -782,6 +782,599 @@ typedef struct _RocdecHevcIQMatrix {
...
+     */
+    int32_t wmmat[8];
+
+    /* valid or invalid on affine set */
+    uint8_t invalid;
+
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t reserved[4];
+
+} RocdecWarpedMotionParamsAV1;
```
Comment (jeffqjiangNew) [inline api/rocdecode.h]: Similar suggestion as above.

File: api/rocdecode.h [hunk 6]
```diff
@@ -782,6 +782,599 @@ typedef struct _RocdecHevcIQMatrix {
...
+    uint8_t cb_mult;
+    uint8_t cb_luma_mult;
+    uint16_t cb_offset;
+    uint8_t cr_mult;
+    uint8_t cr_luma_mult;
+    uint16_t cr_offset;
+
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t reserved[4];
+
```
Comment (jeffqjiangNew) [inline api/rocdecode.h]: Consider removing the empty line here.

File: api/rocdecode.h [hunk 7]
```diff
@@ -782,6 +782,599 @@ typedef struct _RocdecHevcIQMatrix {
...
+    /** \brief indicates if a feature is enabled or not.
+     *  Each bit field itself is the feature_id. Index is segment_id.
+     *  feature_mask[segment_id] & (1 << feature_id) equal to 1 specify that the feature of
+     *  feature_id for segment of segment_id is enabled, otherwise disabled.
+     */
+    uint8_t feature_mask[8];
+
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t reserved[4];
+
```
Comment (jeffqjiangNew) [inline api/rocdecode.h]: Consider removing the empty line here

File: api/rocdecode.h [hunk 8]
```diff
@@ -782,6 +782,599 @@ typedef struct _RocdecHevcIQMatrix {
...
+     *  https://aomedia.googlesource.com/aom/+/refs/heads/master/av1/decoder/decodeframe.c
+     */
+    int32_t wmmat[8];
+
+    /* valid or invalid on affine set */
+    uint8_t invalid;
+
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t reserved[4];
+
```
Comment (jeffqjiangNew) [inline api/rocdecode.h]: Consider removing the empty line here

File: api/rocdecode.h [hunk 9]
```diff
@@ -782,6 +782,596 @@ typedef struct _RocdecHevcIQMatrix {
...
+    /** simplified affine with rotation + zoom only, 4-parameter */
+    RocdecAv1TransformationRotzoom = 2,
+    /** affine, 6-parameter */
+    RocdecAv1TransformationAffine = 3,
+    /** transformation count */
+    RocdecAv1TransformationCount
+} RocdecAv1TransformationType;
+
+typedef struct _RocdecAv1WarpedMotionParams {
+
```
Comment (jeffqjiangNew) [inline api/rocdecode.h]: Remove this empty line as well.

**Review by jeffqjiangNew — APPROVED** [review]

**Review by AryanSalmanpour — APPROVED** [review]

**Pull request merged**